src/Nethermind/Nethermind.Runner/Ethereum/Steps/LoadAnalyticsPlugins.cs

```diff
@@ -0,0 +1,126 @@
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Nethermind.Grpc;
+using Nethermind.Logging;
+using Nethermind.Runner.Ethereum.Context;
+using Nethermind.TxPool.Analytics;
+
+namespace Nethermind.Runner.Ethereum.Steps
+{
+    [RunnerStepDependencies(typeof(InitializeNetwork))]
+    public class LoadAnalyticsPlugins : IStep
+    {
+        private readonly EthereumRunnerContext _context;
+        private readonly ILogger _logger;
+
+        public LoadAnalyticsPlugins(EthereumRunnerContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _logger = context.LogManager.GetClassLogger();
+        }
+
+        private class LogDataPublisher : IDataPublisher
+        {
+            private readonly ILogger _logger;
+
+            public LogDataPublisher(ILogManager logManager)
+            {
+                _logger = logManager.GetClassLogger();
+            }
+
+            public void Publish<T>(T data) where T : class
+            {
+                if (data == null)
+                {
+                    return;
+                }
+
+                if (_logger.IsInfo) _logger.Info(data.ToString());
+            }
+        }
+
+        private class GrpcLogPublisher : IDataPublisher
+        {
+            private readonly IGrpcServer _grpcServer;
+            private readonly ILogger _logger;
+
+            public GrpcLogPublisher(IGrpcServer grpcServer, ILogManager logManager)
+            {
+                _grpcServer = grpcServer ?? throw new ArgumentNullException(nameof(grpcServer));
+                _logger = logManager.GetClassLogger();
+            }
+
+            public void Publish<T>(T data) where T : class
+            {
+                if (data == null)
+                {
+                    return;
+                }
+
+                if (_logger.IsWarn) _logger.Warn($"Publishing data {data.ToString()}");
+                _grpcServer.PublishAsync<T>(data, "analytics");
+
+                if (_logger.IsInfo) _logger.Info(data.ToString());
+            }
+        }
+
+        public virtual Task Execute()
+        {
+            _logger.Warn("Loading analytics plugins");
+            IInitConfig initConfig = _context.Config<IInitConfig>();
+            IGrpcConfig grpcConfig = _context.Config<IGrpcConfig>();
+
+            foreach (string path in Directory.GetFiles(initConfig.PluginsDirectory))
+            {
+                if (path.EndsWith("dll"))
+                {
+                    _logger.Warn($"Loading assembly {path}");
+                    Assembly assembly = Assembly.LoadFile(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, path));
```

Comment:
**Loading generic dll's without strong name to main AppDomain is a security risk**

Reply:
this is our plugin, doesn't matter